src/topp/SageAdapter.cpp

```diff
@@ -90,6 +96,598 @@ class TOPPSageAdapter :
+          {
+             hist[deltamass] += 1.0;
+             num_charges_at_mass[deltamass] += 1.0; 
+             charge_states[deltamass].push_back(h.getCharge()); 
+          }
+        }
+
+        else
+        {
+          hist[deltamass] += 1.0;
+          bool chargecheck = false; 
+          for (auto& x : charge_states[deltamass])
+          {
+            if(x == h.getCharge())
+            {
+              chargecheck = true; 
+            }
+          }
+          if(!chargecheck)
+          {
+            num_charges_at_mass[deltamass] += 1.0; 
+            charge_states[deltamass].push_back(h.getCharge()); 
+          } 
+        }
+      }
+    } */
+      
+
+  
+
+    pair< vector<double>, pair<mapRatetoMass, map<double, double>>> results; 
+    results.second.first = hist; 
+    results.second.second = num_charges_at_mass; 
+    results.first = sorted_keys; 
+    bool with_smoothing = smoothing; 
+
+    if (with_smoothing)
+    {
+      std::map<double, double> smoothed_hist =  smoothMassSpectrum(hist, 0.00001); 
+      cout << "Size of smoothed hist " <<  smoothed_hist.size() << std::endl ; 
+      /* for (auto& x : smoothed_hist)
+      {
+      if(x.second > 5) cout << "First val" << x.first << "Second val" << x.second << std::endl; 
+      } */
+
+
+
+      std::map<double, double> smoothedMaxes2 = findPeaks( smoothed_hist ); 
+      std::map<double, double> smoothedMaxes3 = findPeaks( smoothed_hist, 0.0, 3.0 ); 
+
+      mapRatetoMass num_charges_at_mass_smoothed; 
+
+      /* cout << "Size of smoothed maxes2 " << smoothedMaxes2.size() << std::endl ; 
+      for (auto& x : smoothedMaxes2)
+      {
+        cout << "First val" << x.first << "Second val" << x.second << std::endl; 
+      }
+ */
+      //cout << "Size of smoothed maxes3 " << smoothedMaxes2.size() << std::endl ; 
+      for (auto& x : smoothedMaxes3)
+      {
+        //cout << "First val" << x.first << "Second val" << x.second << "Charge: " <<  num_charges_at_mass[x.first] <<std::endl; 
+        num_charges_at_mass_smoothed[x.first] = num_charges_at_mass[x.first]; 
+      }
+
+      pair<vector <double>, pair<mapRatetoMass, map<double,double>>> results_smoothed; 
+      results_smoothed.second.first = smoothedMaxes3; 
+      results_smoothed.second.second = num_charges_at_mass_smoothed; 
+      results_smoothed.first = sorted_keys; 
+
+      return results_smoothed; 
+
+    }
+    
+    return results ; 
+  }
+
+ vector<PeptideIdentification> mapDifftoMods( mapRatetoMass hist, map<double, double> charge_hist, vector<PeptideIdentification>& pips, double precursor_mass_tolerance_ = 5, bool precursor_mass_tolerance_unit_ppm = true, String outfile = "", vector<double> keys_sorted = {})
+{
+    
+  vector<vector<PeptideIdentification>> clusters(hist.size(), vector<PeptideIdentification>());
+
+  // Accessing the .obo file 
+  ControlledVocabulary unimod_; 
+  try
+  {
+    unimod_.loadFromOBO("PSI-MS", File::find("/CV/unimod.obo"));
```

Comment:
instead of parsing unimod.obo you can directly get all modifications from ModificationsDB.

Reply:
Alright great I will try that out and then test if the results are still the same, thanks for the suggestion.

Reply:
Would the idea here be to also remove the map for the mods and also call functions from ModificationsDB or just to fill the map with ModificationsDB?